examples/nature_of_code/chp_03_oscillation/3_oop_wave_particles.rs

```diff
@@ -0,0 +1,129 @@
+// The Nature of Code
+// Daniel Shiffman
+// http://natureofcode.com
+//
+// example 3-x: OOP Wave Particle
+extern crate nannou;
+
+use nannou::prelude::*;
+
+fn main() {
+    nannou::app(model, event, view).run();
+}
+
+struct Model {
+    wave0: Wave,
+    wave1: Wave,
+}
+
+struct Particle {
+    position: Vector2<f32>,
+}
+
+struct Wave {
+    x_spacing: f32,       // How far apart should each horizontal position be spaced
+    origin: Vector2<f32>, // Where does the wave's first point start
+    theta: f32,           // Start angle at 0
+    amplitude: f32,       // Height of the wave
+    dx: f32, // Value for incementing X, to be calculated as a function of period and x_spacing
+    _y_values: Vec<f32>, // Using a vector to store the height values for the wave (not entirely necessary)
+    particles: Vec<Particle>,
+}
+
+impl Particle {
+    fn new() -> Self {
+        let position = Vector2::new(0.0, 0.0);
+        Particle { position }
```

Comment:
This represents a position, but a `Vector2` is used rather than a `Point2`? Could probs use `Point2`.